### ISSUE TITLE
実験: 相対reward・Napoleon敗北reward=0でAll-Pass/Citizen比較の競り参加率を再評価する

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#420]
Related: #202, #411, #414, #417
Base branch: `feature/issue-202-non-playing-ai`

## 背景 / 修正

このIssueは **raw rewardではなく、「candidate本人のreward - 他4人の平均reward」の相対rewardで比較する**。

元のIssue本文にあった `EV_BID = pN * 2d` は誤り。Napoleon敗北reward=0でも、Napoleon敗北時には他Citizenがrewardを得るため、candidate Napoleonの相対rewardは負になる。

競り判断は以下で固定する。

- BID側: Napoleonとしての相対期待reward
- PASS側（まだ誰も立っていない）: All-Pass = 0
- PASS側（すでに誰か立っている）: Citizenとしての相対期待reward
- Adjutant / role probability / NapoleonAfterPass / NoContract追加価値は使わない
- Napoleon勝利raw reward = `+2d`
- Napoleon敗北raw reward = `0`

## 相対reward定義

各playerについて

```text
relativeReward = ownRawReward - mean(rawReward of other 4 players)
```

を使う。

### NapoleonとしてBIDした場合

Napoleon勝利時のraw reward:

```text
Napoleon = +2d
Adjutant = +d
Citizen x3 = 0
```

したがってcandidate Napoleonの相対reward:

```text
R_N_win = 2d - d/4 = 7d/4
```

Napoleon敗北時のraw reward:

```text
Napoleon = 0
Adjutant = 0
Citizen x3 = +d
```

したがってcandidate Napoleonの相対reward:

```text
R_N_loss = 0 - 3d/4 = -3d/4
```

Napoleon win probabilityを `pN` とすると:

```text
EV_BID(d) = pN * (7d/4) + (1-pN) * (-3d/4)
```

### CitizenとしてPASSした場合

現在契約値を `d_current`、Napoleon側勝率を `pC_nap` とする。

Napoleon勝利時、candidate Citizenの相対reward:

```text
R_C_nap_win = 0 - (2d + d + 0 + 0)/4 = -3d/4
```

Napoleon敗北時、candidate Citizenの相対reward:

```text
R_C_nap_loss = d - (0 + 0 + d + d)/4 = d/2
```

したがって:

```text
EV_CITIZEN = pC_nap * (-3d_current/4)
           + (1-pC_nap) * (d_current/2)
```

### A. まだ誰も立っていない

PASSした場合の比較基準はAll-Pass / NoContract = 0。

```text
BID if EV_BID > 0
else PASS
```

13で立つ場合、Napoleon敗北raw reward=0なら理論境界は:

```text
pN > 30.0%
```

### B. すでに誰かが立っている

PASSした場合はCitizen相対EVと比較する。

```text
BID if max_legal_bid EV_BID > EV_CITIZEN
else PASS
```

同じ `d=13`・同じ勝率Pで単純比較した場合の理論境界は約:

```text
P > 33.3%
```

実際にはraise側 `d_bid` とcurrent contract `d_current` が異なり得るため、各stateの実値で計算する。

## 目的

1. 相対reward基準でNapoleon敗北raw reward=0を評価する
2. 未入札ではAll-Pass=0、既入札ではCitizen相対EVと比較する
3. opening BID率 / raise BID率を分離して測る
4. hand strength別に競り参加傾向を見る
5. selected relative reward / regret / pairwise / best hitを確認する
6. runtime wiringへ進める水準か判断する

## 使用モデル

- Napoleon win probability: #411 M2
- Citizen Napoleon-side win probability: #414 Citizen N-rollout model

Adjutant関連は使わない。

## Evaluation panel

既存#414/#417 artifactsを可能な限り再利用する。

ただし未入札/既入札を区別できないpanelなら、既存all-legal held-out dataset等からcurrent bidding contextを保持したpanelを作る。

新規NN学習は不要。

## 必須出力

未入札・既入札を必ず別集計する。

| context | states | PASS | BID | strong BID | very-strong BID | selected relative reward | regret |
|---|---:|---:|---:|---:|---:|---:|---:|
| no current bid | ... | ... | ... | ... | ... | ... | ... |
| current bid exists | ... | ... | ... | ... | ... | ... | ... |
| overall | ... | ... | ... | ... | ... | ... | ... |

さらに:
- opening BID率
- raise BID率
- current bid level別raise率
- target distribution 13..19
- suit distribution
- strongest-suit match
- pairwise
- best/top3
- hand-strength別BID率

## Threshold sanity check

必ず数式と実装が一致していることをfixtureで確認する。

Napoleon敗北raw reward=0、d=13の場合:

```text
R_N_win  = +22.75
R_N_loss = -9.75
```

したがってAll-Pass=0との境界は厳密に30%。

`pN > 0なら常にBID` になる実装は誤りなので禁止。

また同一d・同一PでCitizenと比較する単純ケースでは約33.3%が境界になることも確認する。

## Representative hands

最低10〜20例について:

```text
context
hand strength
current bid
best legal BID
pN
relative EV_BID
Citizen pNapWin（既入札のみ）
relative EV_CITIZEN / All-Pass=0
selected action
empirical outcome
```

を出す。

特に30%前後のopening境界、既入札でCitizen比較によりPASS/BIDが分かれる例を含める。

## 非目標

- Adjutant再導入
- reward sweep
- NN再学習
- PPO再学習
- runtime wiring

## Git運用

評価scriptだけならbranch/PR不要、`/tmp`利用可。
恒久評価runnerを追加する場合のみ専用branch/PRを作り、baseは `feature/issue-202-non-playing-ai`。mainにはmergeしない。

## 完了時に必ず報告

- evaluation dataset/panel identity
- no-current-bid / current-bid-exists state数
- 相対reward式とsanity check結果
- opening BID率 / raise BID率
- hand-strength別結果
- selected relative reward / regret / pairwise / best/top3
- target/suit分布
- representative examples
- 30% / 約33.3%の理論境界と実測挙動の整合
- runtime wiringへ進めるか
- branch/PR有無
- mainへmergeしていないこと
